dev_docs/contributing/third_party_dependencies.mdx

```diff
@@ -67,6 +67,57 @@ on Github:
   [Permitted Open Source Licenses list](https://github.com/elastic/open-source/blob/main/elastic-product-policy.md#permitted-licenses-list).
 </DocCallOut>
 
+### Dependency evaluation
+
+1. **Snyk health check**
+    - Ensure a minimum health score is >= 80 in [Snyk Advisor](https://snyk.io/advisor).
+    - If the health score is below 80, a valid business justification must be provided.
+
+2. **Vulnerability assessment**. Check vulnerability reports from Snyk for the exact dependency version, available through [Snyk Advisor](https://snyk.io/advisor).
+    If vulnerabilities are present (production dependencies only):
+
+    - **Critical (CVSS 9.0 - 10.0)**:
+      - **False positive:** Proceed to Step 3.
+      - **Confirmed vulnerability:** Reject dependency.
+
+    - **High (CVSS 7.0 - 8.9)**:
+      - **False positive:** Proceed to Step 3.
+      - **Confirmed vulnerability:** Reject dependency.
+
+    - **Medium/Low (CVSS 0.0 - 6.9)**:
+      - **False positive:** Proceed to Step 3.
+      - **Confirmed vulnerability:** Exception can be granted if the vulnerability is confirmed to be impossible to exploit.
+      
+    **Note:** dev-only dependencies are exempt from these criteria and are acceptable without this evaluation since they cannot affect us in any meaningful way.
+
+3. **Handling false positives**. If a vulnerability is considered to be a false positive:
+    - Clearly document why the vulnerability is considered a false positive.
+    - Submit an exception request following the [Vulnerability Exception Policy and Procedures](https://docs.google.com/document/d/1T0GqyWvnTnFWG4TYTSP_Wt_lW-4HqAIlUNhl0fuaklw/edit?usp=sharing).
+    - The exception request must be created **before merging** the dependency.
+
+4. **Approval**. Document the entire assessment clearly, including:
```

Comment:
cc @elastic/kibana-security for discussion.

I think it would be worthwhile to have a step asking the PR author to briefly explain what the dependency is used for, and why they feel adding a new dependency is the best path forward. (Did they explore other options, such as writing code themselves? Does Kibana already include a similar dependency that could satisfy their use case?)

Reply:
++ on this too. I found in the past, though it is rare in my experience, that we had 2 or 3 deps to support similar functionality in multiple locations of the codebase, and these could be consolidated to a single dep. Also, some deps are dep-heavy themselves, and knowing the exact reason might allow for some feedback or guidance on potential alternatives that might be more lightweight.

Reply:
Agreed. Also, as I mentioned above, we'll need to explain the additional burden of exception requests they'll have to carry in the future if they decide to go this route.